modules/spacelift/admin-stack/root-admin-stack.tf

```diff
@@ -84,7 +84,7 @@ module "root_admin_stack" {
   webhook_enabled                         = try(local.root_admin_stack_config.settings.spacelift.webhook_enabled, var.webhook_enabled)
   webhook_endpoint                        = try(local.root_admin_stack_config.settings.spacelift.webhook_endpoint, var.webhook_endpoint)
   webhook_secret                          = try(local.root_admin_stack_config.settings.spacelift.webhook_secret, var.webhook_secret)
-  worker_pool_id                          = local.worker_pools[var.worker_pool_name]
```

Comment:
It does because the lookup. The `worker_pull_name`	default of `null` causes a failed lookup and if you supply a value, the lookup fails in that case as well.